### PR TITLE
[Access] Unify execution data streaming endpoint

### PR DESCRIPTION
**Pull request by illia-malachyn**
#5558

Added handlers for new endpoints such as:
- SubscribeExecutionDataFromStartBlockID
- SubscribeExecutionDataFromStartBlockHeight
- SubscribeExecutionDataFromLatest

This PR depends on https://github.com/onflow/flow/pull/1451
Link to PR for emulator: https://github.com/onflow/flow-emulator/pull/639

### CODE REVIEW

**Comment by illia-malachyn** [comment]
https://github.com/onflow/flow/pull/1451 Should be reviewd firstly

File: engine/access/state_stream/backend/handler.go [hunk 1]
```diff
@@ -88,38 +102,80 @@ func (h *Handler) SubscribeExecutionData(request *executiondata.SubscribeExecuti
...
+	// check if the maximum number of streams is reached
+	if h.StreamCount.Load() >= h.MaxStreams {
+		return status.Errorf(codes.ResourceExhausted, "maximum number of streams reached")
+	}
+	h.StreamCount.Add(1)
+	defer h.StreamCount.Add(-1)
 
-		execData, err := convert.BlockExecutionDataToMessage(resp.ExecutionData)
+	startBlockID := flow.ZeroID
+	if request.GetStartBlockId() != nil {
```
Comment (peterargue) [inline engine/access/state_stream/backend/handler.go:126]: let's remove this conditional. This endpoint should require a valid start block is passed.

File: engine/access/state_stream/backend/handler.go [hunk 2]
```diff
@@ -259,6 +315,37 @@ func (h *Handler) SubscribeEventsFromLatest(request *executiondata.SubscribeEven
...
+		}
+
+		err = convert.BlockExecutionDataEventPayloadsToVersion(execData, eventEncodingVersion)
+		if err != nil {
+			return status.Errorf(codes.Internal, "could not convert execution data event payloads to JSON: %v", err)
+		}
+
+		err = send(&executiondata.SubscribeExecutionDataResponse{
+			BlockHeight:        resp.Height,
+			BlockExecutionData: execData,
```
Comment (peterargue) [inline engine/access/state_stream/backend/handler.go:342]: let's also include the block timestamp, it was missing before
https://github.com/onflow/flow/blob/bfe58c88bcb5945c796bfed8eff17b9222ab3fe4/protobuf/flow/executiondata/executiondata.proto#L251-L261

**Review by peterargue — COMMENTED** [review]
thanks for the PR @illia-malachyn! added a couple small comments, but otherwise this looks good

File: integration/go.mod [hunk 3]
```diff
@@ -361,3 +361,7 @@ require (
 replace github.com/onflow/flow-go => ../
 
 replace github.com/onflow/flow-go/insecure => ../insecure
+
+replace github.com/onflow/flow-emulator v0.62.2-0.20240418140508-d969ff66d9cd => github.com/The-K-R-O-K/flow-emulator v0.61.2-0.20240424125735-9315d02891a9
+
+replace github.com/onflow/flow/protobuf/go/flow v0.4.1-0.20240412170550-911321113030 => github.com/The-K-R-O-K/flow/protobuf/go/flow v0.0.0-20240423092305-0fd55a54a5a7
```
Comment (peterargue) [inline integration/go.mod:367]: you can update this to `v0.4.1`

File: go.mod [hunk 4]
```diff
@@ -320,3 +320,5 @@ require (
 
 // Using custom fork until https://github.com/onflow/flow-go/issues/5338 is resolved
 replace github.com/ipfs/boxo => github.com/onflow/boxo v0.0.0-20240201202436-f2477b92f483
+
+replace github.com/onflow/flow/protobuf/go/flow v0.4.1-0.20240412170550-911321113030 => github.com/The-K-R-O-K/flow/protobuf/go/flow v0.0.0-20240423092305-0fd55a54a5a7
```
Comment (peterargue) [inline go.mod:324]: you can update this to `v0.4.1`

File: integration/go.mod [hunk 5]
```diff
@@ -356,3 +356,5 @@ require (
 replace github.com/onflow/flow-go => ../
 
 replace github.com/onflow/flow-go/insecure => ../insecure
+
+replace github.com/onflow/flow-emulator v0.62.2-0.20240418140508-d969ff66d9cd => github.com/The-K-R-O-K/flow-emulator v0.61.2-0.20240507111935-77ea08a5808d
```
[the remark this answers is not in the document]
Comment (illia-malachyn) [reply integration/go.mod]: I used `13dcdc35eecb5b6fcaa8e2e3a24c101a5e753e30`, which is the next commit (the merge one). Is that fine?

File: integration/go.mod [hunk 6]
```diff
@@ -356,3 +356,5 @@ require (
 replace github.com/onflow/flow-go => ../
 
 replace github.com/onflow/flow-go/insecure => ../insecure
+
+replace github.com/onflow/flow-emulator v0.62.2-0.20240418140508-d969ff66d9cd => github.com/onflow/flow-emulator v0.62.2-0.20240507141305-13dcdc35eecb
```
Comment (peterargue) [inline integration/go.mod:360]: let's remove this `replace` and update the dependency version on line 26

**Review by peterargue — APPROVED** [review]

**Pull request merged**